stm/asyncTaskQueue.mli

```diff
@@ -147,7 +147,7 @@ module MakeQueue(T : Task) () : sig
   (** [create n] will initialize the queue with [n] workers. If [n] is
       0, the queue won't spawn any process, working in a lazy local
       manner. [not imposed by the this API] *)
-  val create : int -> queue
+  val create : int -> CoqworkmgrApi.priority -> queue
```

Comment:
comments are outdated

Reply:
Thanks, I overlooked it. I added `[pri]` in the description of the function.

About this, I have actually a more general question. Why not to pass all of `cur_opt : stm_opt` functionally in the stm. It seems to me that this would make reasoning on the code simpler, showing more explicitly when it is modified or not. (I suspect that it is in fact not modified.)

Also, I found some references to `-toploop` here and there (precisely in `async_proofs_is_master` and `fakeide.ml`) and I guess that it is obsolete comment. Can I remove them?

Even more generally, one of my motivation was to understand if we could simplify the massaging of command line option names at the level of strings in `AsyncTaskQueue.spawn`. At some time I thought we could send the whole `Coqargs.t` directly over the channel as the initial `opts` field of the coqworkers but this would require the type `Coqargs.t` to be known already from the `stm`, i.e. to typically define it in `vernac`. Would it be too demanding? I don't see otherwise how we could pass the command line state to the coq worker (somehow, massaging the string in `asyncTaskQueue.ml` is an evidence that something of the command line argument structure has to be known at the level of the stm!)